### ISSUE TITLE
Does sendRate drop if I stop sending?

### ISSUE THREAD

**Issue by jan-ivar** [w3c/webtransport#476]
The spec isn't clear whether sendRate is a measure of what's occurring now or a prediction:
 
1. [[[SendRate]]](https://w3c.github.io/webtransport/#rate-control-feedback-internal-slots): says: _"The rate at which the [WebTransport](https://w3c.github.io/webtransport/#webtransport) dequeues and sends data, ..."_
2. [sendRate](https://w3c.github.io/webtransport/#dom-webtransportratecontrolfeedback-sendrate) says:_"This is the rate at which queued data will be dequeued and sent by the user agent, in bits per second."_

The first sounds like the rate at which the WebTransport dequeues and sends data _**right now**_, whereas the latter suggests it's the rate data _**will be**_ dequeued, which seems like a prediction.

If this is a prediction, how is it made? Embedded in this question are a few more:

1. Nullable aside, what is the initial sendRate?
2. When does it appear?
3. What is it if nothing is being sent?
4. Once non-null, can it go back to null? What should an app do in that case? 

Basically, lots of questions around how this number is to behave over time from creation onward.

**Comment by pthatcher** [comment]
It's the rate at which data is being sent right now and in the future, until it changes, and then it will have a new value.   If we use the "right now" language, then that's probably more accurate.  The initial sendRate is whatever the browser decides it to be.   The browser may still have a value even if nothing is being sent.  I suppose it could go from non-null to null if there were some kind of major network event that reset congestion control, like a switch between using WiFi and Cellular. 

I suppose if the value is null a web app would do whatever it is doing right now without the signal existing: do its best to guess.

**Comment by jan-ivar** [comment]
Meeting:
- jesup: best estimate you can get from the CC on what the rate will be when you send again
- victor: CC strives to avoid bw drop from burts, bad if you lost bw every time.
- UAs can increase sensitivity after drop
- Can the rate grow above what it is currently if you're not sending at that rate?
- pthatcher: Don't send more than this to avoid queuing (jesup: too strong a guarantee?). Might be useful for app to know what rate is too low it might cause estimate to stop growing
- likely value
- Is estimate affected by pooled traffic? Yes, and covered in Privacy section in previous PR
- victor: once non-null stay non-null
- there is a privacy concern: in some cases some browsers would might a bw estimate and some other stats carried over from previous connections from other origins, related to retaining rtt number to achieve zero rtt
- what if browser detects connection is busted for 5 seconds?
- maybe zero instead of null? null = no idea, some sense. What should the app do with that? 
- keep null at any time.
-

**Comment by pthatcher** [comment]
I have a new PR for adding it to stats: https://github.com/w3c/webtransport/pull/494

**Comment by jan-ivar** [comment]
Meeting:
- most questions resolved by #494 
- Regarding original question, cc could do probing or maybe choose to decay it.
- Knowledge of state of connection gets fuzzier over time. Lots of answers to how to model that best
- Internals we don't want to mandate in the spec
- Add notes? UA should try to ...? Add note recognizing difficulties above? Hard problem here.
- Values may not be accurate if you haven't sent recently

**Comment by jan-ivar** [comment]
My remaining concern is that [estimatedSendRate](https://w3c.github.io/webtransport/#dom-webtransportconnectionstats-estimatedsendrate) is entirely implementation defined number the browsers are free to set to any value. Is there more we can say to ensure interop on this value?

**Comment by wilaw** [comment]
Stopping sending is just one condition under which estimatedSendRate may return a different value across user-agent implementations. Traffic ramp rate, traffic peak, and the issues raised by #559 etc may also effect the answer. This issue scope should be broadened to pose the question "What can we do in the spec to ensure that estimatedSendRate values returned by different UA are reasonably similar?"
